Core/HLE/sceKernelHeap.cpp

```diff
@@ -110,18 +110,65 @@ static u32 SysMemForKernel_536AD5E1()
 	return 0;
 }
 
+static int sceKernelFreeHeapMemory(int heapId, u32 block) {
+	u32 error;
+	KernelHeap* heap = kernelObjects.Get<KernelHeap>(heapId, error);
+	if (heap) {
+		if (block == 0)
+			INFO_LOG(SCEKERNEL, "sceKernelFreeHeapMemory(%d): heapId, block", heapId, 0);
+			return 0;
+		if (!heap->alloc.FreeExact(block)) {
+			ERROR_LOG(SCEKERNEL, "sceKernelFreeHeapMemory(%d): heapId, block Invalid pointer ", heapId, block);
+			return SCE_KERNEL_ERROR_INVALID_POINTER;
+		}
+		return hleLogSuccessInfoI(SCEKERNEL, 0, "sceKernelFreeHeapMemory(%d): heapId, block", heapId, block);;
+	}
+	else {
+		return hleLogError(SCEKERNEL, error, "sceKernelFreeHeapMemory(%d): invalid heapId", heapId);
+	}
+}
+
+static int sceKernelAllocHeapMemoryWithOption(int heapId, u32 memSize, u32 paramsPtr) {
+	u32 error;
+	KernelHeap* heap = kernelObjects.Get<KernelHeap>(heapId, error);
+	if (heap) {
+		u32 grain = 4;
+		// 0 is ignored.
+		if (paramsPtr != 0) {
+			u32 size = Memory::Read_U32(paramsPtr);
+			if (size < 8) {
+				ERROR_LOG(HLE, "sceKernelAllocHeapMemoryWithOption(%08x, %08x, %08x): invalid param size", heapId, memSize, paramsPtr);
+				return 0;
+			}
+			if (size > 8) {
+				WARN_LOG(HLE, "sceKernelAllocHeapMemoryWithOption(): unexpected param size %d", size);
+			}
+			grain = Memory::Read_U32(paramsPtr + 4);
+			INFO_LOG(HLE, "sceKernelAllocHeapMemoryWithOption(%08x, %08x, %08x)", heapId, memSize, paramsPtr);
+			// There's 8 bytes at the end of every block, reserved.
+			memSize += 8;
+			u32 addr = heap->alloc.AllocAligned(memSize, grain, grain, true);
+			return addr;
+		}
```

Comment:
There's no return when `paramsPtr == 0`.  It shouldn't ever not return, it needs to return an error or success or something.

-[Unknown]